refactor/authentication

BassantM opened the pull request:
# What does this PR do?
<!-- Brief description of changes -->

## Type of Change
- [ ] 🐛 Bug fix
- [ ] ✨ New feature
- [x] 🔧 Refactoring
- [ ] 🎨 UI changes

## Changes Made
- [x] Compose UI
- [x] Data
- [x] Domain
- [ ] Koin DI
- [ ] Ktor API

## Testing
- [x] Tested locally
- [x] Unit tests added/updated
- [x] UI tests added/updated

## Screenshots
<!-- Add if UI changes -->

## Checklist
- [x] Code builds without warnings
- [x] Self-reviewed
- [x] Tests pass
- [x] Ready for review

BassantM committed:
refactor:
- authentication repository implementations
- rename some functions
- simplify logic
- move some functions to its correct core, it wasn't include to auth section

BassantM committed:
refactor: update login validation method calls

BassantM committed:
refactor: authentication use case

BassantM committed:
refactor: login view model

BassantM committed:
refactor: url handling logic in registration view model

BassantM committed:
refactor: registration screen

BassantM committed:
refactor: move unrelated tests from authentication repository to account repository test

BassantM committed:
test: registration view model test

BassantM committed:
test: login view model test

BassantM committed:
refactor: update package for registration view model test

BassantM committed:
refactor: delete empty package

BassantM committed:
refactor: move session creation and save outside success check

karimdeabes commented on data/src/main/java/com/london/data/remote/model/authentication/LoginRequestResponse.kt:
```diff
@@ -4,7 +4,7 @@ import kotlinx.serialization.SerialName
 
 @Serializable
-data class LoginValidationRequestBody(
+data class LoginRequestResponse(
```
this is n't response from api , it's post body

karimdeabes commented on data/src/main/java/com/london/data/remote/model/authentication/TokenResponse.kt:
```diff
@@ -4,7 +4,7 @@ import kotlinx.serialization.SerialName
 
 @Serializable
-data class Token(
+data class TokenResponse(
```
same here, not response , check it's usage

BassantM committed:
refactor:
- rename token response to request token
 - rename login request response to login request

karimdeabes reviewed (commented):
in loginScreen when i clicked forget Password and return to the login screen forget Password button doesn't  work

AsmaaKhaledGomaa commented:
When you click on create account, work after clicking many times, + issue Karim mentioned
https://github.com/user-attachments/assets/c45e6a96-71a0-4dfe-a725-57f591ac40ac

AsmaaKhaledGomaa commented on data/src/test/java/com/london/data/repository/AuthenticationRepositoryImplTest.kt:
```diff
@@ -202,8 +202,8 @@ class AuthenticationRepositoryImplTest {
     // region: logout()
     @Test
     fun `logout clears auth and returns true when session exists`() = runTest {
```
Please rename the test function to
function name should return ... when onSaveClick ...

AsmaaKhaledGomaa commented on domain/src/test/kotlin/com/london/domain/usecase/login/AuthenticationUseCaseTest.kt:
```diff
@@ -12,53 +12,53 @@ import org.junit.Test
     }
 
     @Test
     fun `should return true when login as guest succeeds`() = runTest {
```
Please rename the test function to:
Function name should return ... when onSaveClick ...

BassantM replied:
i didn't got what is the wrong of naming @AsmaaKhaledGomaa

BassantM committed:
refactor: when click on forget password and navigate back, can go to forget password again

BassantM committed:
refactor: when click on forget password and navigate back, can go to forget password again

BassantM committed:
refactor: rename fetchAndSaveUserAccount to getUserAccount
